### ISSUE TITLE
Rename doesn't work when composing multiple case classes

### ISSUE THREAD

**Issue by hejfelix** [stanch/reftree#11]
So I'm generating these two images from the code below:

"integer.png"
![integer](https://user-images.githubusercontent.com/1153154/27259526-6578b916-5415-11e7-9591-897bf0608e87.png)

"application_with_integer"
![application_with_integer](https://user-images.githubusercontent.com/1153154/27259528-68b1ac1e-5415-11e7-9d93-d86c81753e7c.png)

In the first case, my integer case class is correctly renamed to bob. In the second case, no renaming happens.

```scala
implicit val tint: ToRefTree.DerivationConfig[Integer] =
  ToRefTree
    .DerivationConfig[Integer]
    .rename("bob")
    .omitField("i")

val renderer = Renderer(
  renderingOptions = RenderingOptions(density = 75),
  animationOptions = AnimationOptions(interpolationDuration = 3.seconds, framesPerSecond = 1),
  directory = Paths.get("images")
)
renderer.render("application_with_integer", Diagram(Application(Integer(42), Identifier("x"))))
renderer.render("integer", Diagram(Integer(42)))
```

I am not sure if I am using the API correctly, but I can see that the "i" field is omitted in both cases, i.e. some of the options must be picked up in both cases.

**Comment by stanch** [comment]
Thanks! Reproduced like so:

```scala
sealed trait AST
case class Integer(i: Int) extends AST
case class Application(tree: AST, method: String) extends AST
  
implicit val c = ToRefTree.DerivationConfig[Integer].rename("Bob").omitField("i")

render(Application(Integer(8), "x"))
```

Probably something is wrong in the coproduct derivation code.

**Comment by stanch** [comment]
Actually, `render(Integer(8): AST)` is enough to reproduce this.

**Comment by stanch** [comment]
@hejfelix Just released `1.1.0-20170618` with a fix, could you try it?

**Comment by hejfelix** [comment]
fixed 👍
